mixins/scrollBetweenSections.ts

```diff
@@ -50,12 +58,17 @@ export default class extends Vue {
       const triggerWindowBottom = rootBounds.bottom
       const onTop = targetTop >= 0 && targetTop <= triggerWindowBottom
       if (onTop && targetTop < highestTopValue) {
-        this.activeSection = target.id
+        if (this.activeSection !== target.id) {
+          this.activeSection = target.id
+          this.activeSectionChanged()
+        }
         highestTopValue = targetTop
       }
     })
   }
 
+  activeSectionChanged () {}
```

Comment:
Probably adding a small JSDoc here would be beneficial to avoid ambiguity at first glance.

From the name of the method, it could also be interpreted as a function that performs a check to see if the active section has changed and returns the evaluation of that check.

To be fair, after checking the `void` return type and the implementation in the extendable class, it's clear what it does, but still, some cognitive load could be saved with something like this:

```suggestion
  /**
   * This methods gets called when the active section changes.
   */
  activeSectionChanged () {}
```

Reply:
I was thinking of multiple implementations such as an event or a variable callback. I thought that is the most simple implementation but I can change it.
I think that writing some documentation will be good anyway.

Reply:
I added a comment on InfinteScroll also to make it easier to understand.

Reply:
As an alternative renaming the method to something like `activeSectionChangedCallback` or `activeSectionChangedCb` could also remove said ambiguity IMO.